pandas-stubs/io/excel/_base.pyi

```diff
@@ -112,7 +111,8 @@ def read_excel(  # type:ignore[misc]
     dtype_backend: DtypeBackend | NoDefault = ...,
 ) -> dict[str, DataFrame]: ...
 @overload
-def read_excel(
+# mypy says this won't be matched
+def read_excel(  # type: ignore[misc]
```

Comment:
Could maybe use bound=Int|Str for IntStrT to potentially get rid of this last overload?

Reply:
the bound option would make the 3rd overload obsolete for pyright (it passes all tests) but mypy still needs the 3rd overload.